Adds fallback for 1.25 clients

lazypower opened the pull request:
This adds a fetch from configured url option to fetch a dockerbeat release. Defaults to the current upstream release candidate.

Must contain a matching SHA1 sum, otherwise the download will fail and the charm will be in a blocked state.

Commit:
Adds fallback for 1.25 clients

This adds a fetch from configured url option to fetch a dockerbeat release. Defaults to the current upstream release candidate.

Must contain a matching SHA1 sum, otherwise the download will fail and the charm will be in a blocked state.

marcoceppi commented:
:+1: I'd suggest we make a note to remove this fall back option when 2.0 is gold (or maybe 2.0.1). But at least having a revision in the store for a bit longer which does both will be decent.

Commit:
Adds notes about resource delivery

Adds notes about resoure delivery and multi-arch concerns. Additionally adds notes about the compat behavior baked in for 1.25 deployments.

mbruzek commented:
I hate to add more config options that we will remove later (making upgrades harder), especially these that are _ahem_ **immutable**. Could we hardcode the fallback url and hashsum instead?

mbruzek commented on README.md:
```diff
@@ -79,6 +79,18 @@ To monitor additional hosts, simply relate the Dockerbeat subordinate
+## DockerBeat Delivery
+
+This charm makes use of Resources. A juju 2.0 feature. When deploying this
+charm on a juju 2.0 enabled controller, the upstream ingensi dockerbeat
```
Capital i? According to this: https://github.com/Ingensi it is a company and always seems to be capitalized in github.

mbruzek commented on README.md:
```diff
@@ -79,6 +79,18 @@ To monitor additional hosts, simply relate the Dockerbeat subordinate
+### 1.25 Compatibility
+Alternatively, on juju 1.25 hosts, this charm supports a configurable URL and
+SHA1 sum configuration option to attempt to fetch. Configured for a release
+from the ingensi github repository.
```
Capital i?

lazypower commented:
@mbruzek i'm not hard coding any values like that in here, no thanks. thats a time :bomb: waiting to happen.

Commit:
Make config mutable....again

Commit:
Capitalize Ingensi

mbruzek commented:
+1
